plugins/flashrom/fu-plugin-flashrom.c

```diff
@@ -205,15 +286,76 @@ fu_plugin_flashrom_startup(FuPlugin *plugin, GError **error)
 		return FALSE;
 	}
 	flashrom_set_log_callback(fu_plugin_flashrom_debug_cb);
+
+	if (flashrom_programmer_init(&data->flashprog, "internal", NULL)) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "programmer initialization failed");
+		return FALSE;
+	}
+
+	rc = flashrom_flash_probe(&data->flashctx, data->flashprog, NULL);
+	if (rc == 3) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "flash probe failed: multiple chips were found");
+		return FALSE;
+	}
+	if (rc == 2) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "flash probe failed: no chip was found");
+		return FALSE;
+	}
+	if (rc != 0) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "flash probe failed: unknown error");
+		return FALSE;
+	}
+
 	return TRUE;
 }
 
+static gboolean
+fu_plugin_flashrom_unlock(FuPlugin *self, FuDevice *device, GError **error)
```

Comment:
out of interest; why here and not in fu-flashrom-device.c?

Reply:
Unlock operation is defined only at the plugin level, at least I don't see corresponding function for a device. Or do you suggest to create `fu_flashrom_device_unlock()` and call it here?

Reply:
> to create fu_flashrom_device_unlock() and call it here

Yes, I think that's a good idea. Then if we do a FuDevice->unlock() vfunc in the future we can connect it up easily.